src/components/templates/Camp/hooks/useKeyboardControls.ts

```diff
@@ -0,0 +1,219 @@
+
+  const downPress = useKeyPress("ArrowDown");
+  const upPress = useKeyPress("ArrowUp");
+  const leftPress = useKeyPress("ArrowLeft");
+  const rightPress = useKeyPress("ArrowRight");
+  const enterPress = useKeyPress("Enter");
+
+  const roomEnter = useCallback(
+    (room: CampRoomData) => {
+      const roomVenue = venues?.find((venue) =>
+        room.url.endsWith(`/${venue.id}`)
+      );
+      const venueRoom = roomVenue
+        ? { [roomVenue.name]: currentTimeInUnixEpoch }
+        : {};
+      room &&
+        user &&
+        enterRoom(
+          user,
+          {
+            [`${venue.name}/${room.title}`]: currentTimeInUnixEpoch,
+            ...venueRoom,
+          },
+          profile?.lastSeenIn
+        );
+    },
+    [profile, user, venue, venues]
+  );
+
+  // TODO: this is duplicated logic from handlePotentialRoomClick in Map component, so we can probably simplify
+  // TODO: do we want to filter this? Or just handle 1 room hit
+  const hitRoom = useCallback(
+    (row: number, column: number) => {
+      let isHitting = false;
+      venue.rooms.forEach((room: CampRoomData) => {
+        const rowPosition = (100 / rows) * row;
+        const colPosition = (100 / columns) * column;
+        const roomX = Math.round(room.x_percent);
+        const roomY = Math.round(room.y_percent);
+        const roomWidth = Math.round(room.width_percent);
+        const roomHeight = Math.round(room.height_percent);
+
+        if (
+          rowPosition >= roomY &&
+          rowPosition <= roomY + roomHeight &&
+          colPosition >= roomX &&
+          colPosition <= roomX + roomWidth
+        ) {
+          setSelectedRoom(room);
+          setIsHittingRoom(true);
+          isHitting = true;
+        } else {
+          if (isHittingRoom && selectedRoom === room) {
+            setSelectedRoom(undefined);
+            setIsHittingRoom(false);
+            isHitting = false;
+          }
+        }
+      });
+      return isHitting;
+    },
+    [
+      columns,
+      isHittingRoom,
+      venue.rooms,
+      rows,
+      selectedRoom,
+      setIsHittingRoom,
+      setSelectedRoom,
+    ]
+  );
+
+  useEffect(() => {
+    if (!venue.id) return;
+
+    if ((!currentPosition?.row && !currentPosition?.column) || keyDown) {
+      return;
+    }
+
+    const { row, column } = currentPosition;
+    if (row && column) {
+      const seatTaken = (r: number, c: number) => partygoersBySeat?.[r]?.[c];
+      if (enterPress && selectedRoom) {
+        setKeyDown(true);
+        setTimeout(() => setKeyDown(false), MOVEMENT_INTERVAL);
+        openRoomUrl(selectedRoom.url);
+        roomEnter(selectedRoom);
+        return;
+      }
+      if (downPress) {
+        setKeyDown(true);
+        setTimeout(() => setKeyDown(false), MOVEMENT_INTERVAL);
+        if (row + 1 > DEFAULT_ROWS || seatTaken(row + 1, column)) {
+          return;
+        }
+        hitRoom(row + 1, column);
+        takeSeat(row + 1, column);
```

Comment:
These are some more examples of the arbitrary `+ 1` logic mentioned earlier, so if we can clean that up to be `0-indexed, then we can make all of this simpler too.